lib/types/query/query.ts

```diff
@@ -26,26 +26,27 @@ export type SysQueries<Sys extends FieldsType> = ExistenceFilter<Sys, 'sys'> &
   SubsetFilters<Sys, 'sys'> &
   RangeFilters<Sys, 'sys'>
 
-export type EntryFieldsQueries<Fields extends FieldsType = FieldsType> =
-  | EntrySelectFilter<Fields>
+export type EntryFieldsQueries<Fields extends FieldsType> =
+  | EntrySelectFilterWithFields<Fields>
   | ExistenceFilter<Fields, 'fields'>
-  | (EqualityFilter<Fields, 'fields'> & InequalityFilter<Fields, 'fields'>)
+  | EqualityFilter<Fields, 'fields'>
+  | InequalityFilter<Fields, 'fields'>
   | FullTextSearchFilters<Fields, 'fields'>
   | SubsetFilters<Fields, 'fields'>
   | LocationSearchFilters<Fields, 'fields'>
   | RangeFilters<Fields, 'fields'>
 
 // TODO: create-contentful-api complained about non-optional fields when initialized with {}
-export type EntriesQueries<Fields extends FieldsType = FieldsType> = EntryFieldsQueries<Fields> &
-  SysQueries<Pick<EntrySys, 'createdAt' | 'updatedAt' | 'revision' | 'id' | 'type'>> &
-  FixedQueryOptions &
-  FixedPagedOptions & { content_type?: string } & {
-    resolveLinks?: never
-  } & { order?: string }
+export type EntriesQueries<Fields extends FieldsType> =
+  | (EntryFieldsQueries<Fields> & { content_type: string })
```

Comment:
`content_type` should be allowed to be used as the only query as well, and I think line this only allows it to be paired with entry queries.

Reply:
I think considering the fact most field queries are optional then that's included in this case

Reply:
All should be optional apart from `content_type`. This just says that you can’t use `EntryFieldsQueries` without `content_type`. If there is another property that is required, then we should make it optional.

Reply:
@msieroslawska There is one important detail that I learned recently that may not be obvious. If you have a type like `{ a?: string } | { b?: string }`, you’ll be able to combine the properties from the two parts of the union because `{}` matches both.  `{a: '', b: '' }` doesn’t look like it would match the type but it does. The behavior only changes when you have required properties.